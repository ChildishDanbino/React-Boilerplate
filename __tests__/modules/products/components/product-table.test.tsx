import React from 'react';
import { render, fireEvent } from 'test-utils'
import '@testing-library/jest-dom/extend-expect';
import ProductTable from 'modules/products/components/product-table';

describe('ProductTable', () => {
    it('Renders ProductTable Componnent when valid data passed', () => {
        const mockProductData = [
            {
                "category": 1,
                "description": "With a philosophical flourish Cato throws himself upon his sword; I quietly take to the ship.",
                "id": 734,
                "logo": "https://via.placeholder.com/64",
                "name": "Bio-ove",
                "quantity": 13
            },
        ];

        const { asFragment } = render(<ProductTable products={mockProductData} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Renders null when no products', () => {
        const { queryByTestId } = render(<ProductTable products={[]} />);
        expect(queryByTestId('buy-now')).toBeNull()
    });

    it('Alerts the user when Buy Now Clicked', () => {
        window.alert = jest.fn();
        const mockProductData = [
            {
                "category": 1,
                "description": "With a philosophical flourish Cato throws himself upon his sword; I quietly take to the ship.",
                "id": 734,
                "logo": "https://via.placeholder.com/64",
                "name": "Bio-ove",
                "quantity": 13
            },
        ];

        const { getByText } = render(<ProductTable products={mockProductData} />);
        expect(getByText('Buy Now')).toBeInTheDocument();
        fireEvent.click(getByText("Buy Now"));
        //@ts-ignore
        expect(window.alert.mock.calls.length).toBe(1);
    });
});
