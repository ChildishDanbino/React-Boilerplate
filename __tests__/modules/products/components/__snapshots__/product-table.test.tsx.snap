// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTable Renders ProductTable Componnent when valid data passed 1`] = `
<DocumentFragment>
  <div
    class="sc-bwzfXH iyXQWR"
  >
    <table
      class="sc-htpNat KtFsv"
    >
      <tbody>
        <tr
          class="sc-bxivhb eqWZMb"
        >
          <th
            class="sc-ifAKCX iCNBXQ"
          >
            NAME
          </th>
          <th
            class="sc-ifAKCX iCNBXQ"
          >
            DESCRIPTION
          </th>
          <th
            class="sc-ifAKCX iCNBXQ"
          >
            PHOTO
          </th>
          <th
            class="sc-ifAKCX iCNBXQ"
          >
            QUANTITY
          </th>
          <th
            class="sc-ifAKCX iCNBXQ"
          >
            BUY NOW
          </th>
        </tr>
        <tr
          class="sc-bxivhb eqWZMb"
        >
          <td
            class="sc-EHOje eWHMDo"
          >
            Bio-ove
          </td>
          <td
            class="sc-EHOje eWHMDo"
          >
            With a philosophical flourish Cato throws himself upon his sword; I quietly take to the ship.
          </td>
          <td
            class="sc-EHOje eWHMDo"
          >
            <img
              alt="product_image"
              src="https://via.placeholder.com/64"
            />
          </td>
          <td
            class="sc-EHOje eWHMDo"
          >
            13
          </td>
          <td
            class="sc-EHOje eWHMDo"
          >
            <button
              class="sc-bdVaJa cIoHTR"
            >
              Buy Now
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
